### PR TITLE
[ie/pornhub] Update login to email

### LINKED ISSUE #9717: Login broken for PornHub premium
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

USA

### Provide a description that is worded well enough to be understood

Pornhub recently updated their login page to only login using email instead of username. Since this change it is now not possible to login (even when using known good credentials).

I tracked down a login form field name change that is causing the issue, and was able to fix it by updating the field name. 

In the yt_dlp->extractor->pornhub.py _login method, the login form fields that are updated should be updated from `login_form.update({'username' : username,'password': password,})` to `login_form.update({'email' : username,'password': password,})`.

I am not super familiar with github so sorry this isn't a pull request. 

P.S. verbose log attached is an example of pre-update failure. After the update I can confirm it worked. 

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
"C:\Program Files\Python312\python.exe" C:\Dev\pythonProject3\main.py 
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version stable@2023.12.30 from yt-dlp/yt-dlp [f10589e34] (pip) API
[debug] params: {'forceprint': {'video': ['filename']}, 'username': '#####HIDDEN INTENTIONALLY#####', 'password': '#####HIDDEN INTENTIONALLY#####', 'verbose': True, 'writethumbnail': True, 'embedthumbnail': True, 'postprocessors': [{'key': 'FFmpegMetadata', 'add_metadata': True}, {'key': 'EmbedThumbnail'}], 'outtmpl': {'default': 'C:\\Group playlist downloaded\\2024-04-18\\%(title)s_%(id)s.%(ext)s'}, 'updatetime': True, 'sleep_interval': 5, 'max_sleep_interval': 15, 'retry_sleep_functions': {'fragment': 60, 'http': 5, 'file_access': 1}, 'retries': 500, 'fragment_retries': 10, 'file_access_retries': 500, 'abort_on_unavailable_fragments': True, 'concurrent_fragment_downloads': 2, 'http_chunk_size': 25000000, 'buffersize': 1000000000000, 'throttledratelimit': 4000000, 'download_archive': 'D:\\archive.txt', 'overwrites': False, 'compat_opts': set(), 'http_headers': {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'Accept-Language': 'en-us,en;q=0.5', 'Sec-Fetch-Mode': 'navigate'}}
[debug] Python 3.12.2 (CPython AMD64 64bit) - Windows-11-10.0.22635-SP0 (OpenSSL 3.0.13 30 Jan 2024)
[debug] exe versions: ffmpeg 6.1.1-full_build-www.gyan.dev (setts), ffprobe 6.1.1-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2023.11.17, mutagen-1.47.0, requests-2.31.0, sqlite3-3.43.1, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1798 extractors
[debug] Loading archive file 'D:\\archive.txt'
[PornHub] Extracting URL: https://www.pornhubpremium.com/view_video.php?viewkey=65bff9e374ae2
[PornHub] Downloading pornhubpremium login page
[PornHub] Logging in to pornhubpremium
ERROR: [PornHub] 65bff9e374ae2: Unable to login: Invalid entry. Use a valid email address to log in.
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\extractor\common.py", line 718, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\extractor\pornhub.py", line 273, in _real_extract
    self._login(host)
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\extractor\pornhub.py", line 117, in _login
    raise ExtractorError(

Traceback (most recent call last):
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\YoutubeDL.py", line 1587, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\YoutubeDL.py", line 1722, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\extractor\common.py", line 718, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\extractor\pornhub.py", line 273, in _real_extract
    self._login(host)
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\extractor\pornhub.py", line 117, in _login
    raise ExtractorError(
yt_dlp.utils.ExtractorError: [PornHub] 65bff9e374ae2: Unable to login: Invalid entry. Use a valid email address to log in.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Dev\pythonProject3\main.py", line 109, in <module>
    ydl.download(urls)
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\YoutubeDL.py", line 3541, in download
    self.__download_wrapper(self.extract_info)(
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\YoutubeDL.py", line 3516, in wrapper
    res = func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\YoutubeDL.py", line 1576, in extract_info
    return self.__extract_info(url, self.get_info_extractor(key), download, extra_info, process)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\YoutubeDL.py", line 1605, in wrapper
    self.report_error(str(e), e.format_traceback())
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\YoutubeDL.py", line 1054, in report_error
    self.trouble(f'{self._format_err("ERROR:", self.Styles.ERROR)} {message}', *args, **kwargs)
  File "C:\Users\Andrew\AppData\Roaming\Python\Python312\site-packages\yt_dlp\YoutubeDL.py", line 993, in trouble
    raise DownloadError(message, exc_info)
yt_dlp.utils.DownloadError: ERROR: [PornHub] 65bff9e374ae2: Unable to login: Invalid entry. Use a valid email address to log in.

Process finished with exit code 1
```

### PR DESCRIPTION
**Pull request by feederbox826**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Update PornHub login to use `email` field instead of `username`

Fixes #9717

Again, no tests were added since a dummy account and matching playlist ID are required

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/pornhub.py [hunk 1]
```diff
@@ -77,8 +77,8 @@ def _login(self, host):
         # on both sites so that we have to identify them as the same video.
         # For that purpose we have to keep both in the same extractor
         # but under different netrc machines.
-        username, password = self._get_login_info(netrc_machine=site)
-        if username is None:
+        email, password = self._get_login_info(netrc_machine=site)
+        if email is None:
```
Comment (bashonly) [inline yt_dlp/extractor/pornhub.py:81]: I would only change the form field
```suggestion
        username, password = self._get_login_info(netrc_machine=site)
        if username is None:
```

File: yt_dlp/extractor/pornhub.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def is_logged(webpage):
         login_form = self._hidden_inputs(login_page)
 
         login_form.update({
-            'username': username,
+            'email': email,
```
Comment (bashonly) [inline yt_dlp/extractor/pornhub.py:100]: like this
```suggestion
            'email': username,
```

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/pornhub.py (hunk 1 above)
Comment (feederbox826) [reply yt_dlp/extractor/pornhub.py:81]: I set the internal variables to `email` so that they would be more useful in verbose debugging

Comment (bashonly) [reply yt_dlp/extractor/pornhub.py:81]: these lines won't show up in the traceback though

Comment (feederbox826) [reply yt_dlp/extractor/pornhub.py]: OK, will revert

**Review by bashonly — APPROVED** [review]

**Pull request merged**